app/utils/ControllersReferenceData.scala

```diff
@@ -146,16 +146,17 @@ class ControllersReferenceData @Inject()(
         } catch {
           case ex: Exception => {
             Logger.warn(
-              s"[ControllersReferenceData][responseErrorHandler] Could not parse GenericServerError System Error number: ${e4.message}." +
-                s" Showing default error page instead",
+              s"[ControllersReferenceData][responseErrorHandler] Could not parse GenericServerError System Error number: ${if (!e4.message.isEmpty) e4.message else "<no error code>"}. Showing default error page instead",
```

Comment:
Hmm two thoughts, first is just curious whether you went !x.isEmpty instead of nonEmpty for a specific reasoning? (I see it done sometimes so the code reads a certain way).

The other is whether this if statement logic is needed? The way I read it, there shouldn't be a situation where there isn't an exception message. The GenericServerErrorException requires a string in the case class to be instantiated. Which means the only way we get an exception thrown here is when that string doesn't match a key in the messages file for the line 137 comparison.

Reply:
I've added it because of the Kibana logs I've seen: https://kibana.tools.production.tax.service.gov.uk/goto/968d2c8666315c729220f88ac0537d48
I didn't change the functionality - only noted the situations where there doesn't seem to be a code provided there?

only used !isEmpty for out of habit - happy to use .nonEmpty instead if it's a bad habit

Reply:
Ahh empty string scenario, makes sense. Also, using !.isEmpty vs .nonEmpty isn't a hill I'll die on, but I think in most cases .nonEmpty is easier to read.